### PR TITLE
[FLINK-15263][hive][doc] add dedicated page for HiveCatalog

### PR DESCRIPTION
**Pull request by bowenli86**
## What is the purpose of the change

add dedicated page for HiveCatalog with more explanation and example

## Brief change log

add dedicated page for HiveCatalog with more explanation and example

## Verifying this change
n/a

## Does this pull request potentially affect one of the following parts:

n/a

## Documentation

n/a

### CODE REVIEW

**Comment by bowenli86** [comment]
@xuefuz @lirui-apache  @JingsongLi  @zjuwangg

File: docs/dev/table/hive/hive_catalog.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+tom,15
+john,21
+
+{% endhighlight %}
+
+
+#### step 4: start SQL CLI, and create a Kafka table with Flink SQL DDL
+
+Start Flink SQL CLI, create a simple Kafka 2.3.0 table via DDL, and verify its schema.
```
Comment (xuefuz) [inline docs/dev/table/hive/hive_catalog.md]: Flink SQL CLI => Flink SQL CLIENT?

File: docs/dev/table/hive/hive_catalog.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+   'connector.version' = 'universal',
+   'connector.topic' = 'test',
+   'connector.properties.zookeeper.connect' = 'localhost:2181',
+   'connector.properties.bootstrap.servers' = 'localhost:9092',
+   'format.type' = 'csv',
+   'update-mode' = 'append'
+);
+[INFO] Table has been created.
+
+Flink SQL> DESCRIBE test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.md]:
```suggestion
Flink SQL> DESCRIBE mykafka;
```

File: docs/dev/table/hive/hive_catalog.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+ |-- age: INT
+
+{% endhighlight %}
+
+Verify the table from Hive side via Hive Cli:
+
+{% highlight bash %}
+hive> show tables;
+OK
+test
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.md]: ditto

File: docs/dev/table/hive/hive_catalog.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+Time taken: 0.038 seconds, Fetched: 1 row(s)
+
+hive> describe test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.md]: ditto

File: docs/dev/table/hive/hive_catalog.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+{% endhighlight %}
+
+
+#### step 5: run Flink SQL to query the Kakfa table
+
+Run a simple select query from Flink SQL CLI
+
+{% highlight bash %}
+Flink SQL> select * from test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.md]: ditto

File: docs/dev/table/hive/hive_catalog.zh.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+   'connector.version' = 'universal',
+   'connector.topic' = 'test',
+   'connector.properties.zookeeper.connect' = 'localhost:2181',
+   'connector.properties.bootstrap.servers' = 'localhost:9092',
+   'format.type' = 'csv',
+   'update-mode' = 'append'
+);
+[INFO] Table has been created.
+
+Flink SQL> DESCRIBE test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.zh.md]:
```suggestion
Flink SQL> DESCRIBE mykafka;
```

File: docs/dev/table/hive/hive_catalog.zh.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+Verify the table from Hive side via Hive Cli:
+
+{% highlight bash %}
+hive> show tables;
+OK
+test
+Time taken: 0.038 seconds, Fetched: 1 row(s)
+
+hive> describe test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.zh.md]: ditto

File: docs/dev/table/hive/hive_catalog.zh.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+root
+ |-- name: STRING
+ |-- age: INT
+
+{% endhighlight %}
+
+Verify the table from Hive side via Hive Cli:
+
+{% highlight bash %}
+hive> show tables;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.zh.md]: ditto

File: docs/dev/table/hive/hive_catalog.zh.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+
+{% endhighlight %}
+
+
+#### step 5: run Flink SQL to query the Kakfa table
+
+Run a simple select query from Flink SQL CLI
+
+{% highlight bash %}
+Flink SQL> select * from test;
```
Comment (zjuwangg) [inline docs/dev/table/hive/hive_catalog.zh.md]: ditto

File: docs/dev/table/hive/hive_catalog.md [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+## How to use HiveCatalog
+
+Once configured properly, `HiveCatalog` should just work out of box. Users can create Flink meta-objects with DDL, and shoud
+see them immediately afterwards.
+
+### Example
+
+We will walk thru a simple example here.
+
+#### step 1: set up a Hive Metastore
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_catalog.md]: Maybe we can just add links to the Hive docs about how to setup a remote metastore service, given that our Hive integration doesn't require any special configurations of the HMS?

File: docs/dev/table/hive/hive_catalog.md [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+name                	string
+age                 	int
+Time taken: 0.056 seconds, Fetched: 2 row(s)
+
+{% endhighlight %}
+
+
+#### step 5: run Flink SQL to query the Kakfa table
+
+Run a simple select query from Flink SQL CLI
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_catalog.md]: Mention that SQL CLI needs a Flink cluster to run the query, either standalone or yarn-session?

File: docs/dev/table/hive/hive_catalog.md (hunk 2 above)
Comment (bowenli86) [reply docs/dev/table/hive/hive_catalog.md]: good catch!

**Pull request closed without merging**